packages/dds/tree/src/feature-libraries/chunked-forest/chunkTree.ts

```diff
@@ -461,13 +464,18 @@ export function insertValues(
 
 	// Slow path: walk shape and cursor together, inserting values.
 	if (shape.hasValue) {
-		values.push(cursor.value);
+		if(typeof cursor.value === 'string' && idCompressor !== undefined && isStableNodeKey(cursor.value)){
```

Comment:
I think checking `isStableNodeKey(cursor.value)` here is not strictly necessary: tryRecompress should return undefined if the if isn't valid. I think it might be a performance win to do what you did for now, but @taylorsw04  said he would fix that by adding an early out to tryRecompress. Sync up with him on this.

Reply:
I had a chat with Taylor a couple days ago, and he mentioned that calling `isStableId` on the string first is the prescribed way to do this already :)